util/flightcontrol/flightcontrol_test.go

```diff
@@ -223,6 +223,28 @@ func TestContention(t *testing.T) {
 	wg.Wait()
 }
 
+func TestMassiveParallel(t *testing.T) {
+	var retryCount int64
+	g := &Group[string]{}
+	eg, ctx := errgroup.WithContext(context.Background())
+	for i := 0; i < 1000; i++ {
```

Comment:
With old configuration this test reaches 15 second limit ~75% with N=100 and ~25% with N=50

New configuration
N=10000 max-backoff=300ms
N=1000, max-backoff=70ms
N=500,  max-backoff=40ms
N=100,  max-backoff=12ms

---

These numbers are not exactly correct(and would be different on different machines) as they changed when I removed debug and cleaned up some code. The updated code does not fail as easily anymore if N<100 but the relative difference between the reachable backoff is still same.